### ISSUE TITLE
Support settings DJANGO_SETTINGS_MODULE in pytest_configure

### ISSUE THREAD

**Issue by pelme** [pytest-dev/pytest-django#146]
See comment in #119, this should be possible:

```python
import os

def pytest_configure():
    os.environ.setdefault('DJANGO_SETTINGS_MODULE', 'myproject.settings')
```

**Comment by julen** [comment]
> *configuring DJANGO_SETTINGS_MODULE in pytest_configure is not really supported/tested at the moment, but should be easy to support properly*.

How easy is it? I need this to switch to Django 1.7 + pytest-django 2.7, otherwise I'm forced to call `setup()` manually.

**Comment by pelme** [comment]
Actually, I think this works and the issue should be closed. Could you give it a try if you haven't?

There doesn't seem to be a test for this case, so that should probably be added to make sure it doesn't accidentally break.

**Comment by julen** [comment]
I still get the same `INTERNALERROR` traceback:
```python
INTERNALERROR> AssertionError: assert <django.conf.LazySettings object at 0x10eb0e4e0>.configured is True
```

I added a test in #192. If you import `django.setup` and call it after setting `DJANGO_SETTINGS_MODULE`, the test passes.

**Comment by julen** [comment]
Any updates @pelme?

**Comment by blueyed** [comment]
The problem appears to be that pytest-django sets up Django after any project's conftest:

    @pytest.mark.trylast
    def pytest_configure():
        if django_settings_is_configured():
            _setup_django()

([Source](https://github.com/pytest-dev/pytest-django/blob/master/pytest_django/plugin.py#L203-206))

Should be fixed via PR #216.
